src/components/analyses/listing/Listing.js

```diff
@@ -129,6 +133,19 @@ function Listing(props) {
         },
     });
 
+    const [renameAnalysesMutation, { isLoading: renameLoading }] = useMutation(
+        renameAnalyses,
+        {
+            onSuccess: () => {
+                queryCache.invalidateQueries(ANALYSES_LISTING_QUERY_KEY);
+                setRenameDialogOpen(false);
```

Comment:
I wonder if we can move away from reloading the view on mutations like rename, delete etc...since react-query gives us option to do optimistic update. See - https://react-query.tanstack.com/docs/guides/updates-from-mutation-responses
https://react-query.tanstack.com/docs/guides/optimistic-updates
But this can be tried in another PR.

Reply:
Ah yes, good idea 👍 
I forgot about that for rename mutations; but I didn't want to do this for deletes, since the listing can be paged, and removing 1 or more analyses from the listing will change the view for all pages, so I thought it was best to reload the listing in that case.

Reply:
That's a good point.  Wouldn't the same be true for renames as well then? If I have 50 pages and rename `zzzFile` to `aaaFile`, for example, then maybe a reload is still best in that case

Reply:
It seems to me like we should do both things most of the time. For a synchronous rename/delete, we can either update or remove the entry (possibly also remove with a rename or with some renames -- e.g. for the zzzFile->aaaFile thing, if we know that aaaFile wouldn't be on the same page) so that the user gets immediate feedback and initiate a background update of the query that re-updates the display when finished; for async stuff, the same, but based on the 'complete' notification (except if we ever want to do some sort of "task in progress" indicator in listings, which would presumably get put in place immediately for async stuff).

I actually think this might be easier for deletes, because if you're deleting from a page you can largely assume that any newly-visible rows will be at the bottom of the page (page of a, b, c, d; four more after that, if you delete b and d you can show a page of a, c, and two loading skeletons until the query figures out that it should be showing a, c, e, f). With a rename you'd need to reproduce the sorting and make a guess about whether the new name will come before the page (meaning a first-row loading skeleton, probably), after the page (last-row skeleton), or elsewhere within the same page such that it can be updated to that place in the page. (renames are easy if the sorting isn't by name at all, though)

Reply:
For renames I figured it was OK to just update the row in-place, since I don't think the user would be confused if that one row is out of place in the sort, because they were just updating it, and they can reload or navigate to a new page to get the true sort order?

For deletes, where the user could remove half the rows or more, I think it would look worse to leave a bunch of empty space, unless we did something more complicated as described above.

I figured out how to update the listing for a rename, leaving the updated row in-place regardless of the sort order, without refreshing the entire listing. I can push that change to this PR before merging if that's the direction we want to go for now.

Reply:
Hm... not sure.  Seems reasonable to me to keep the renamed value in place, though I'm not sure I've seen that kind of behavior elsewhere. Maybe others have additional thoughts.

As for async, I keep forgetting about that 😖 I did the simplest thing for the DE - I think it refreshes and shows an announcer that the actual action will complete in the background - and then copied it over for deleting files/folders in  Sonora. That's definitely not at all set in stone or ideal, we can make it better in Sonora.